### ISSUE TITLE
after some epochs, training stops

### ISSUE THREAD

**Issue by DilaraAlb** [junyanz/pytorch-CycleGAN-and-pix2pix#703]
After some epochs (178), training stops without giving any error or finishing running. Why that happens, how can I solve?

![image](https://user-images.githubusercontent.com/7215154/61290752-4d675100-a7d5-11e9-8b1d-b83ccfc3093d.png)

**Comment by DilaraAlb** [comment]
On my second trial, it stopped at 186th epoch.

**Comment by junyanz** [comment]
It might be caused by visdom according to this [post](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/issues/619). Could you disable visdom by setting `--display_id 0` and try it again?

**Comment by DilaraAlb** [comment]
It did work, thank you. I have previously checked for similar issues, I missed this one, though.

**Comment by 2110317008** [comment]
(epoch: 70, iters: 838, time: 0.313, data: 0.003) D_A: 0.298 G_A: 0.078 cycle_A: 0.483 idt_A: 0.126 D_B: 0.312 G_B: 0.256 cycle_B: 0.289 idt_B: 0.150
(epoch: 70, iters: 938, time: 0.323, data: 0.007) D_A: 0.307 G_A: 0.221 cycle_A: 0.312 idt_A: 0.174 D_B: 0.245 G_B: 0.262 cycle_B: 0.374 idt_B: 0.194
(epoch: 70, iters: 1038, time: 1.455, data: 0.002) D_A: 0.188 G_A: 0.177 cycle_A: 0.761 idt_A: 0.226 D_B: 0.061 G_B: 0.534 cycle_B: 0.487 idt_B: 0.241
saving the model at the end of epoch 70, iters 76860
End of epoch 70 / 200    Time Taken: 355 sec
learning rate = 0.0002000
(epoch: 71, iters: 40, time: 0.311, data: 0.003) D_A: 0.032 G_A: 0.719 cycle_A: 0.247 idt_A: 0.226 D_B: 0.338 G_B: 0.577 cycle_B: 0.532 idt_B: 0.179
(epoch: 71, iters: 140, time: 0.338, data: 0.007) D_A: 0.248 G_A: 0.932 cycle_A: 0.437 idt_A: 0.199 D_B: 0.105 G_B: 0.685 cycle_B: 0.593 idt_B: 0.213
(epoch: 71, iters: 240, time: 0.298, data: 0.003) D_A: 0.230 G_A: 0.958 cycle_A: 0.405 idt_A: 0.214 D_B: 0.260 G_B: 0.106 cycle_B: 0.477 idt_B: 0.178
(epoch: 71, iters: 340, time: 1.699, data: 0.001) D_A: 0.233 G_A: 0.502 cycle_A: 0.303 idt_A: 0.149 D_B: 0.179 G_B: 0.243 cycle_B: 0.299 idt_B: 0.138
(epoch: 71, iters: 440, time: 0.307, data: 0.003) D_A: 0.273 G_A: 0.463 cycle_A: 0.316 idt_A: 0.172 D_B: 0.227 G_B: 0.242 cycle_B: 0.329 idt_B: 0.139
(epoch: 71, iters: 540, time: 0.308, data: 0.002) D_A: 0.070 G_A: 0.477 cycle_A: 0.292 idt_A: 0.156 D_B: 0.219 G_B: 0.808 cycle_B: 0.344 idt_B: 0.136
(epoch: 71, iters: 640, time: 0.309, data: 0.001) D_A: 0.151 G_A: 0.409 cycle_A: 0.295 idt_A: 0.150 D_B: 0.224 G_B: 0.386 cycle_B: 0.366 idt_B: 0.153

在maps数据集上用cycle第一次训练，在这个地方停下不动了？能帮帮解决下吗？非常谢谢！

**Comment by junyanz** [comment]
Have you run `python -m visdom.server` before running the training script? You can also disable visdom by setting `--display_id 0`.

**Comment by SsnL** [comment]
@2110317008 @DilaraAlb What's the stack trace if you just interrupt when it hangs?

**Comment by 2110317008** [comment]
我设置--display_id 0后就可以了！谢谢你们！
但是又出现了一个新的问题是epoch一直等于1？

**Comment by 2110317008** [comment]
> Have you run `python -m visdom.server` before running the training script? You can also disable visdom by setting `--display_id 0`.

您好！我想问您一个问题，当我用自己的数据集A作为trainA数据集B作为trainB时，epoch一直等于1？

**Comment by 2110317008** [comment]
之前epoch一直等于1，我就把trainA和trainB里面的图片减少了几百张（之前近万张），然后epoch就正常了。
对于训练数据集（trainA and trainB）,w我阅读了下面的、之前公布的
How can I make it work for my own data？ 
#309 320 202
The current code only supports RGB and grayscale images. If you would like to train the model on other data types, please follow the following steps:
change the parameters --input_nc and --output_nc to the number of channels in your input/output images.
Write your own custom data loader (It is easy as long as you know how to load your data with python).  If you write a new data loader class, you need to change the flag --dataset_mode accordingly.  
Alternatively, you can modify the existing data loader. For aligned datasets, change this line; For unaligned datasets, change these two lines.
If you use visdom and HTML to visualize the results, you may also need to change the visualization code.
trainA和trainB里面的数据图片一定要一样多？里面的数据图片有一定的数量规定吗？
